src/cpp/CMakeLists.txt

```diff
@@ -33,6 +27,12 @@ endif()
 file(GLOB_RECURSE sources "*.cpp" "*.h")
 add_library(cpp ${sources})
 
+set_source_files_properties(
+        ${sources}
+        PROPERTIES
+        OBJECT_DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/cprover_library.inc"
+)
```

Comment:
This should just be `add_dependency(cpp "${CMAKE_CURRENT_BINARY_DIR}/cprover_library.inc")` I think?

Reply:
Copied from ansi-c, will try updating both.

Reply:
This doesn't work:

```
CMake Error at src/cpp/CMakeLists.txt:30 (add_dependencies):
  The dependency target
  "/home/tkiley/workspace/cbmc/build/debug/src/cpp/cprover_library.inc" of
  target "cpp" does not exist.
```

You're more than welcome to raise a follow up PR making this nicer, but I'd rather just merge this as is, rather than spend more time on it, since it does fix the problem of the C++ library and macros in the C library not working on CMake.